mountpoint-s3/src/inode/negative_cache.rs

```diff
@@ -0,0 +1,204 @@
+use std::time::{Duration, Instant};
+
+use linked_hash_map::LinkedHashMap;
+
+use super::{expiry::Expiry, InodeNo};
+
+use crate::sync::RwLock;
+
+/// A caches for negative lookups.
+/// Maintains a bounded set of (parent_ino, child_name) entries that expire after a fixed time.
+#[derive(Debug)]
+pub struct NegativeCache {
+    /// Holds keys in expiration order from oldest to newest.
+    map: RwLock<LinkedHashMap<Key, Expiry>>,
```

Comment:
so this is going to be `100'000` entities max and we expect it to use no more than `20MB` of space, any particular estimations on how much memory a single entity could take? the key is generally unbounded, right?

Reply:
The rough estimation is based on indirectly measuring Mountpoint process memory while trying to lookup random entries (I used UUIDs, 37 characters): the result of the test is a marginal cost of <200 bytes per entry.
In general, it will depend on the length of the key, but it is not really unbounded: consider that the max file name length is 255 bytes.

Reply:
S3 supports keys up to 1024 bytes. Since this is a negative cache, I'm sure that a user could ask for something even longer but we can assume 1024 as a sensible upper bound.

I think it would help to document how we chose 100,000 as the default max size, and maybe some of the reasoning here can go into it.

Reply:
> I'm sure that a user could ask for something even longer but we can assume 1024 as a sensible upper bound.

seems like they can not given the linux limitation for the file size, right?

adjusting Alex's estimate (with 37 chars for names) for names max of 255 chars, we'll have `< 420 bytes` per entity => `420 * 100'000 / 1024 / 1024 ~< 40 MB` in total would be more precise, but yeah I see the values matching in order of magnitude :)